### PR TITLE
Adding f.pipe

### PR DESCRIPTION
**Pull request by brodeynewman**

### CODE REVIEW

File: packages/pipe/src/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * After the first function in the pipe chain completes, that value is then passed to the next
+ * function in the pipe chain.
+ * @param {Array<function>} arguments - functions to pipe data through
+ * @returns {*} - whatever is returned from the pipe chain
+ * @example
+ * const add = (a, b) => a + b;
+ * const mult = a => a * a;
+ *
+ * const piped = pipe(add, mult)
+ * piped(2, 3) // 25;
```
Comment (JasonPollman) [inline packages/pipe/src/index.js]: Been doing `// => 25` everywhere, for consistency.

Comment (brodeynewman) [reply packages/pipe/src/index.js]: okay

File: packages/pipe/src/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (typeof arguments[argLen] !== 'function') {
+      throw new TypeError(TYPE_ERROR);
+    }
+  }
+
+  return function receiver() {
+    let value = funcs[0].apply(this, arguments);
+    let len = 0;
+
+    while (++len < funcs.length) {
```
Comment (JasonPollman) [inline packages/pipe/src/index.js]: Might be faster to do `while(++len < argLen)`.

Comment (brodeynewman) [reply packages/pipe/src/index.js]: you right. I refactored a bit and forgot I had a variable for funcs.length

File: packages/pipe/src/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      const func = funcs[len];
+
+      value = func.call(this, value);
```
Comment (JasonPollman) [inline packages/pipe/src/index.js]: Unnecessary variable. `value = funcs[len].call(this, value)`.

Comment (brodeynewman) [reply packages/pipe/src/index.js]: fair enough

**Pull request merged**